cpp/include/raft/neighbors/detail/cagra/search_multi_kernel.cuh

```diff
@@ -993,5 +983,71 @@ struct search : search_plan_impl<DATA_T, INDEX_T, DISTANCE_T, SAMPLE_FILTER_T> {
+          class DISTANCE_T_,
+          class INDEX_T_,
+          typename SAMPLE_FILTER_T>
+struct search<TEAM_SIZE,
+              DATASET_BLOCK_DIM,
+              cagra_q_dataset_descriptor_t<DATA_T,
+                                           CODE_BOOK_T,
+                                           PQ_BITS,
+                                           PQ_CODE_BOOK_DIM,
+                                           0,
+                                           DISTANCE_T_,
+                                           INDEX_T_,
+                                           0>,
+              SAMPLE_FILTER_T>
+  : public search_plan_impl<cagra_q_dataset_descriptor_t<DATA_T,
+                                                         CODE_BOOK_T,
+                                                         PQ_BITS,
+                                                         PQ_CODE_BOOK_DIM,
+                                                         0,
+                                                         DISTANCE_T_,
+                                                         INDEX_T_,
+                                                         0>,
+                            SAMPLE_FILTER_T> {
+  using DATASET_DESCRIPTOR_T = cagra_q_dataset_descriptor_t<DATA_T,
+                                                            CODE_BOOK_T,
+                                                            PQ_BITS,
+                                                            PQ_CODE_BOOK_DIM,
+                                                            0,
+                                                            DISTANCE_T_,
+                                                            INDEX_T_,
+                                                            0>;
+  using INDEX_T              = typename DATASET_DESCRIPTOR_T::INDEX_T;
+  using DISTANCE_T           = typename DATASET_DESCRIPTOR_T::DISTANCE_T;
+
+  search(raft::resources const& res,
+         search_params params,
+         int64_t dim,
+         int64_t graph_degree,
+         uint32_t topk)
+    : search_plan_impl<DATASET_DESCRIPTOR_T, SAMPLE_FILTER_T>(res, params, dim, graph_degree, topk)
+  {
+  }
+
+  void set_params(raft::resources const& res) {}
+
+  void operator()(raft::resources const& res,
+                  DATASET_DESCRIPTOR_T dataset_desc_,
+                  raft::device_matrix_view<const INDEX_T, int64_t, row_major> graph,
+                  INDEX_T* const topk_indices_ptr,       // [num_queries, topk]
+                  DISTANCE_T* const topk_distances_ptr,  // [num_queries, topk]
+                  const DATA_T* const queries_ptr,       // [num_queries, dataset_dim]
+                  const uint32_t num_queries,
+                  const INDEX_T* dev_seed_ptr,              // [num_queries, num_seeds]
+                  uint32_t* const num_executed_iterations,  // [num_queries,]
+                  uint32_t topk,
+                  SAMPLE_FILTER_T sample_filter)
+  {
+  }
```

Comment:
This is the placeholder for the CAGRA-Q specialization of the multi_kernel search kernels, right? Since the quantized version is not yet implemented, we could probably throw a "not implemented" error, and this feature can be added in next release.

Reply:
Added error message output in search class constructor